test/e2e/network/service.go

```diff
@@ -3733,84 +3687,64 @@ func validateEndpointsPorts(c clientset.Interface, namespace, serviceName string
 	return nil
 }
 
-// restartApiserver restarts the kube-apiserver.
-func restartApiserver(namespace string, cs clientset.Interface) error {
-	// TODO: Make it work for all providers.
-	if !framework.ProviderIs("gce", "gke", "aws") {
-		return fmt.Errorf("unsupported provider for RestartApiserver: %s", framework.TestContext.Provider)
-	}
-	if framework.ProviderIs("gce", "aws") {
-		initialRestartCount, err := getApiserverRestartCount(cs)
-		if err != nil {
-			return fmt.Errorf("failed to get apiserver's restart count: %v", err)
-		}
-		if err := sshRestartMaster(); err != nil {
-			return fmt.Errorf("failed to restart apiserver: %v", err)
-		}
-		return waitForApiserverRestarted(cs, initialRestartCount)
-	}
-	// GKE doesn't allow ssh access, so use a same-version master
-	// upgrade to teardown/recreate master.
-	v, err := cs.Discovery().ServerVersion()
+// restartComponent restarts component static pod
+func restartComponent(cs clientset.Interface, cName, ns string, labelSet labels.Set) error {
+	pods, err := getComponentPod(cs, ns, labelSet)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get %s's pods, err: %v", cName, err)
 	}
-	return framework.MasterUpgradeGKE(namespace, v.GitVersion[1:]) // strip leading 'v'
+	if err := killComponentPod(cs, ns, pods); err != nil {
```

Comment:
This won't work for GKE (whereas the previous MasterUpgradeGKE works).

Reply:
yeah, we can not see kube-apiserver int gke, but we still can see kube-proxy pod.

Reply:
but this changes breaks things that works before this PR

Reply:
GKE this scene should be considered  in this PR.

Reply:
So if in AWS, can we use the approach of `killComponentPod ` to restart kube-apiserver. I find that case calculate kube-apiserver count by search as pod.

https://github.com/kubernetes/kubernetes/blob/ac10e4043af29c651407add4ead61975471f2fbd/test/e2e/network/service.go#L3730

https://github.com/kubernetes/kubernetes/blob/ac10e4043af29c651407add4ead61975471f2fbd/test/e2e/network/service.go#L3784-L3793

And if no pod, it will return err. And the case will failed.

Reply:
Let me clarify what I meant one more time.

Currently (without your PR), if run on GKE it would work, because we call MasterUpgradeGKE.

After your change - this will no longer work for GKE.

So this is a regression.
Please fix to not regress for the GKE usecase.

Reply:
All right, I fix it first.

Reply:
@wojtek-t ,thanks for reminding! In GKE  scene, it will do as before.

Reply:
Thanks!